### PR TITLE
MLOPS-87 wanna managed-notebook sync --force

### PR DESCRIPTION
**Pull request by JacekHebda**

### CODE REVIEW

File: src/wanna/core/services/notebook.py [hunk 1]
```diff
@@ -696,14 +694,14 @@ def sync(self) -> None:
         if to_be_created:
             to_be_created_str = "\n".join(["-" + item.name for item in to_be_created])
             logger.user_info(f"Managed notebooks to be created:\n{to_be_created_str}")
-            should_create = typer.confirm("Are you sure you want to create them?")
-            if should_create:
+            should_create = typer.confirm("Are you sure you want to create them?") if not force else False
```
Comment (jsilva) [inline src/wanna/core/services/notebook.py:697]: I would do `should_create = True if force else typer.confirm("Are you sure you want to create them?")`

and then remove the `force or` in the next line

same for should_delete

WDYT

Comment (JacekHebda) [reply src/wanna/core/services/notebook.py:697]: Nice logic optimisation :) I will correct

File: tests/notebook/test_service.py [hunk 2]
```diff
@@ -195,3 +195,8 @@ def test_build(self):
         config = load_config_from_yaml("samples/notebook/managed-notebook/wanna.yaml", "default")
         nb_service = ManagedNotebookService(config=config, workdir=Path("."))
         assert nb_service.build() == 0
+
+    def test_sync(self):
+        config = load_config_from_yaml("samples/notebook/managed-notebook/wanna.yaml", "default")
+        nb_service = ManagedNotebookService(config=config, workdir=Path("."))
+        assert nb_service.sync(force=True) == 0
```
Comment (jsilva) [inline tests/notebook/test_service.py:202]: I am not sure what is being tested here, besides the fact that doesn't throw errors. 

What about returning `to_be_created` and `to_be_deleted` and assert on the output. This way we can also check that the _diff functions does its job

The current assert could the `nb_service.sync(force=True) == None`(assuming no `return 0/1` is introduced)  and the effect would be the same

**Review by jsilva — CHANGES_REQUESTED** [review]

File: src/wanna/core/services/notebook.py [hunk 3]
```diff
@@ -642,12 +642,12 @@ def _get_jupyterlab_link(self, instance_id: str) -> str:
         instance_info = self.notebook_client.get_runtime({"name": instance_id})
         return f"https://{instance_info.access_config.proxy_uri}"
 
-    def _return_diff(self):
+    def _return_diff(self) -> Tuple[List[Any], List[Any]]:
```
Comment (jsilva) [inline src/wanna/core/services/notebook.py:645]: Is not returning Any, is returning some object, please add the correct type. thx

**Review by jsilva — CHANGES_REQUESTED** [review]

File: tests/notebook/test_service.py (hunk 2 above)
Comment (JacekHebda) [reply tests/notebook/test_service.py:202]: Not throwing errors is already sth, as it made me change the code a bit (the .runtimes part) :)
Asserting == None is impossible in our setup, hence introducting the return 0/1
I will add the assertion on to_be_created and to_be_deleted then

**Review by jsilva — CHANGES_REQUESTED** [review]

File: tests/notebook/test_service.py (hunk 2 above)
Comment (jsilva) [reply tests/notebook/test_service.py:202]: `result = nb_service.sync(force=True)`
and 

`self.assertIsNone(result)` is possible

**Pull request merged**